pennylane_lightning/lightning_kokkos/_adjoint_jacobian.py

```diff
@@ -0,0 +1,339 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Internal methods for adjoint Jacobian differentiation method.
+"""
+
+try:
+    from pennylane_lightning.lightning_kokkos_ops.algorithms import (
+        AdjointJacobianC64,
+        AdjointJacobianC128,
+        create_ops_listC64,
+        create_ops_listC128,
+    )
+except ImportError:
+    pass
+
+from os import getenv
+from typing import List
+
+import numpy as np
+import pennylane as qml
+from pennylane import BasisState, QuantumFunctionError, StatePrep
+from pennylane.measurements import Expectation, MeasurementProcess, State
+from pennylane.operation import Operation
+from pennylane.tape import QuantumTape
+
+# pylint: disable=import-error, no-name-in-module, ungrouped-imports
+from pennylane_lightning.core._serialize import QuantumScriptSerializer
+from pennylane_lightning.core.lightning_base import _chunk_iterable
+
+from ._state_vector import LightningKokkosStateVector
+
+
+class LightningKokkosAdjointJacobian:
+    """Check and execute the adjoint Jacobian differentiation method.
+
+    Args:
+        qubit_state(LightningKokkosStateVector): State Vector to calculate the adjoint Jacobian with.
+        batch_obs(bool): If serialized tape is to be batched or not.
+    """
+
+    def __init__(self, kokkos_state: LightningKokkosStateVector, batch_obs: bool = False) -> None:
+        self._qubit_state = kokkos_state
+        self._state = kokkos_state.state_vector
+        self._dtype = kokkos_state.dtype
+        self._jacobian_lightning = (
```

Comment:
How can we get all these codecov warnings? :thinking:

Reply:
I do not know, for some reason, Codecov is not showing the report 😕

However, I ran the test locally and the code is covered. 

For more information about coverage, you can consult my https://github.com/PennyLaneAI/pennylane-lightning/pull/810#issuecomment-2302642149 🚀